File: test/lira_output/py/api_service/memory_store.py
# module api_memory_store
from __future__ import annotations
from .models import User

from .store import UserStore

class InMemoryUserStore(UserStore):
    users: list[User] = []

    def get(self, id: str) -> User | None:
        for user in self.users:
            if (user.id == id):
                return user
        return None


    def save(self, user: User) -> User:
        self.users.append(user)
        return user


    def list(self) -> list[User]:
        return self.users

File: test/lira_output/ts/api_service/memory_store.ts
// module api_memory_store
import { User } from "./models";

import { UserStore } from "./store";

export class InMemoryUserStore implements UserStore {
  private users: User[] = [];
  public get(id: string): User | null {
    for (const user of this.users) {
      if ((user.id == id)) {
        return user;
      }
    }
    return null;
  }
  public save(user: User): User {
    this.users.push(user);
    return user;
  }
  public list(): User[] {
    return this.users;
  }
}
